### PR TITLE
Unlock ImageIO disks after importer failure.

### PR DESCRIPTION
**Pull request by mrnold**
**What this PR does / why we need it**:
This pull request unlocks ImageIO source disks when the importer returns an error. The importer was not finalizing the transfer through the ImageIO API because the finalize code lives in the Close method, which previously relied on a "defer" that was skipped by os.Exit. In the linked bug, the transfer was successful, but the resize failed, triggering an os.Exit that skipped the deferred Close and therefore the finalize.

This pull request adds an explicit call to Close just before the relevant calls to os.Exit (there are only a few). In the interest of further safety, this pull request also cancels a transfer if the finalize fails.

**Which issue(s) this PR fixes**:
Fixes [BZ#1945522](https://bugzilla.redhat.com/show_bug.cgi?id=1945522)

**Release note**:
```release-note
Unlock ImageIO disks when transfer returns an error.
```

### CODE REVIEW

File: cmd/cdi-importer/importer.go [hunk 1]
```diff
@@ -190,12 +190,14 @@ func main() {
 		if err != nil {
 			klog.Errorf("%+v", err)
 			if err == importer.ErrRequiresScratchSpace {
+				dp.Close()
```
Comment (maya-r) [inline cmd/cdi-importer/importer.go:193]: Why is this needed if we already have `defer dp.Close()` up above?

Comment (maya-r) [reply cmd/cdi-importer/importer.go:193]: Apparently runtime.Goexit exists to be os.Exit while running deferred calls.

Comment (mrnold) [reply cmd/cdi-importer/importer.go:193]: Thanks, that would be much cleaner. It looks like it's meant to be called from a goroutine though, I will run some tests and see how well it works here.

Comment (mrnold) [reply cmd/cdi-importer/importer.go:193]: Goexit does not appear to be a drop-in replacement, it hangs after the transfer if I just use it in place of os.Exit. It does unlock the disk though, so it must be running the deferred Close correctly.

Comment (maya-r) [reply cmd/cdi-importer/importer.go:193]: thanks for trying
